src/fluentexpr.prw

```diff
@@ -18,10 +18,20 @@ Static Function Format( cString, aValues )
     Local cResult := cString
     Local nIndex
     For nIndex := 1 To Len( aValues )
-        cResult := StrTran( cResult, '{' + AllTrim( Str( nIndex ) ) + '}', cValToChar( aValues[ nIndex ] ) )
+        cResult := StrTran( cResult, '{' + AllTrim( Str( nIndex ) ) + '}', ToString( aValues[ nIndex ] ) )
     Next
     Return cResult
 
+Static Function ToString( xValue )
+    If ValType( xValue ) == "A"
```

Comment:
`ValType` will be called for each check.
`Local cType := ValType( xValue )` and `Do Case` is the way to go.